src/fondant/compiler.py

```diff
@@ -121,15 +121,24 @@ def _generate_spec(
 
         pipeline.validate(run_id=run_id)
 
+        component_cache_key = None
+
         for component_name, component in pipeline._graph.items():
             component_op = component["fondant_component_op"]
 
+            if component_cache_key is None:
```

Comment:
I think we can remove the `if/else` check here, since we check it inside `get_component_cache_key()`

Reply:
ahh yes, good catch